apps/trackers/jira/query.py

```diff
@@ -447,3 +477,315 @@ def query_comment(self):
+        if severity not in allowed_values:
+            raise MissingSeverityError(
+                f"Jira project {self.ps_module.bts_key} does not have the {field_name} field value "
+                f"{severity}; allowed values are: {', '.join(allowed_values)}"
+            )
+        self._query["fields"][field_id] = {"value": severity}
+
+    # TODO write tests - tracked in OSIDB-2980
+    def generate_source(self):
+        field_name = "Source"
+        field = JiraProjectFields.objects.filter(
+            project_key=self.ps_module.bts_key, field_name=field_name
+        ).first()
+        if field is None:
+            raise MissingVulnerabilityIssueFieldError(
+                f"Field {field_name} not available for Vulnerability issuetype in "
+                f"Jira project {self.ps_module.bts_key}."
+            )
+        allowed_values = field.allowed_values
+        field_id = field.field_id
+        flaw_source = self.tracker.affects.first().flaw.source
+        flaw_source_lower = flaw_source.lower()
+        choice_found = None
+        for choice in allowed_values:
+            if choice.lower() == flaw_source_lower:
+                choice_found = choice
+                break
+        if not choice_found:
+            raise MissingSourceError(
+                f"Jira project {self.ps_module.bts_key} does not have the {field_name} field value "
+                f"{flaw_source}; allowed values are: {', '.join(allowed_values)}"
+            )
+        self._query["fields"][field_id] = {"value": choice_found}
+
+    # TODO write tests - tracked in OSIDB-2980
+    def generate_cve_id(self):
+        field_name = "CVE ID"
+        field = JiraProjectFields.objects.filter(
+            project_key=self.ps_module.bts_key, field_name=field_name
+        ).first()
+        if field is None:
+            raise MissingVulnerabilityIssueFieldError(
+                f"Field {field_name} not available for Vulnerability issuetype in "
+                f"Jira project {self.ps_module.bts_key}."
+            )
+        field_id = field.field_id
+        cves = sorted(
+            [flaw.cve_id for flaw in self.flaws if flaw.cve_id], key=cve_id_comparator
+        )
+        if not cves:
+            raise TrackerCreationError("Tracker can't be filed without a CVE.")
+        if len(cves) > 1:
+            # Multi-CVE flaws not supported by our administrative processes right now.
+            # Supporting that would require checking that OSIDB logic actually works
+            # correctly, and updating logic that uses expressions like
+            # `self.tracker.affects.first().flaw` compatible only with single-flaw trackers.
+            # When that is resolved, either concatenate all CVE IDs with commas,
+            # or change the Jira schema to an array.
+            raise TrackerCreationError(
+                f"Trackers can't be filed for multi-CVE flaws. Attempted creation for CVEs: {repr(cves)}."
+            )
```

Comment:
Aha! So there's theoretically two ways to get a multi-CVE tracker, one of them is currently impossible, and for the other one it should not be my call to disallow it here. Fixing to save multiple CVEs delimited by a comma to the jira field in case there are more of them.